service/app/queries/get_blob_test.go

```diff
@@ -3,25 +3,107 @@ package queries_test
 import (
 	"testing"
 
+	"github.com/boreq/errors"
 	"github.com/planetary-social/go-ssb/di"
 	"github.com/planetary-social/go-ssb/fixtures"
 	"github.com/planetary-social/go-ssb/service/app/queries"
+	"github.com/planetary-social/go-ssb/service/domain/blobs"
 	"github.com/stretchr/testify/require"
 )
 
 func TestGetBlob(t *testing.T) {
-	q, err := di.BuildTestQueries()
-	require.NoError(t, err)
-
 	id := fixtures.SomeRefBlob()
+	data := fixtures.SomeBytes()
+
+	testCases := []struct {
+		Name          string
+		Query         queries.GetBlob
+		ExpectedError error
+	}{
+		{
+			Name: "only_id",
+			Query: queries.GetBlob{
+				Id:   id,
+				Size: nil,
+				Max:  nil,
+			},
+			ExpectedError: nil,
+		},
+		{
+			Name: "id_and_correct_size",
+			Query: queries.GetBlob{
+				Id:   id,
+				Size: sizePtr(blobs.MustNewSize(int64(len(data)))),
+				Max:  nil,
+			},
+			ExpectedError: nil,
+		},
+		{
+			Name: "id_and_incorrect_size",
+			Query: queries.GetBlob{
+				Id:   id,
+				Size: sizePtr(blobs.MustNewSize(10)),
+				Max:  nil,
+			},
+			ExpectedError: errors.New("blob size doesn't match the provided size"),
+		},
+		{
+			Name: "id_and_max_above_size",
+			Query: queries.GetBlob{
+				Id:   id,
+				Size: nil,
+				Max:  sizePtr(blobs.MustNewSize(int64(len(data)) + 1)),
+			},
+			ExpectedError: nil,
+		},
+		{
+			Name: "id_and_max_equal_to_size",
+			Query: queries.GetBlob{
+				Id:   id,
+				Size: nil,
+				Max:  sizePtr(blobs.MustNewSize(int64(len(data)))),
+			},
+			ExpectedError: nil,
+		},
+		{
+			Name: "id_and_max_below_size",
+			Query: queries.GetBlob{
+				Id:   id,
+				Size: nil,
+				Max:  sizePtr(blobs.MustNewSize(int64(len(data)) - 1)),
+			},
+			ExpectedError: errors.New("blob is larger than the provided max size"),
+		},
+		{
+			Name: "size_wins_over_max",
+			Query: queries.GetBlob{
+				Id:   id,
+				Size: sizePtr(blobs.MustNewSize(1)),
+				Max:  sizePtr(blobs.MustNewSize(1)),
+			},
+			ExpectedError: errors.New("blob size doesn't match the provided size"),
+		},
+	}
 
-	q.BlobStorage.MockBlob(id, fixtures.SomeBytes())
+	for _, testCase := range testCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			q, err := di.BuildTestQueries()
+			require.NoError(t, err)
 
-	query := queries.GetBlob{
-		Id: id,
+			q.BlobStorage.MockBlob(id, data)
+
+			rc, err := q.Queries.GetBlob.Handle(testCase.Query)
+			if testCase.ExpectedError != nil {
+				require.EqualError(t, err, testCase.ExpectedError.Error())
+				require.Nil(t, rc)
+			} else {
+				require.NoError(t, err)
+				require.NotNil(t, rc)
+			}
+		})
 	}
+}
 
-	rc, err := q.Queries.GetBlob.Handle(query)
-	require.NoError(t, err)
-	require.NotNil(t, rc)
+func sizePtr(s blobs.Size) *blobs.Size {
+	return &s
 }
```

Comment:
Maybe introducing `internal/ptr.go` could turn out handy? I've seen some similar funcs scattered over the codebase (`sizePointer`, `wantDistancePointer` etc.)? 
```
func Ptr[T any](o T) *T {
	return &o
}
```

Reply:
[379abd7](https://github.com/planetary-social/go-ssb/pull/10/commits/379abd7354afec1f3052fe70661b1dabef72b4b5)

Reply:
Ok, cleaned up.